cli/metrics/metadata/build.go

```diff
@@ -0,0 +1,192 @@
+	var bm metadatatypes.Build
+	dockercfg := config.LoadDefaultConfigFile(io.Discard)
+	if alias, ok := dockercfg.Aliases["builder"]; ok {
+		command = alias
+	}
+	if command == "build" {
+		// TODO(@crazy-max): include cli version (e.g., docker;20.10.10)
+		bm.Cli = "docker"
+		bm.Builder = "buildkit"
+		if enabled, _ := isBuildKitEnabled(); !enabled {
+			bm.Builder = "legacy"
+		}
+	} else if command == "buildx" {
+		// TODO(@crazy-max): include buildx version (e.g., buildx;0.6.3)
+		bm.Cli = "buildx"
+		bm.Builder = buildxDriver(dockercfg, args)
+	}
+	return bm
+}
+
+// isBuildKitEnabled returns whether buildkit is enabled either through a
+// daemon setting or otherwise the client-side DOCKER_BUILDKIT environment
+// variable
+func isBuildKitEnabled() (bool, error) {
+	if buildkitEnv := os.Getenv("DOCKER_BUILDKIT"); len(buildkitEnv) > 0 {
+		return strconv.ParseBool(buildkitEnv)
+	}
+	apiClient, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv, dockerclient.WithAPIVersionNegotiation())
+	if err != nil {
+		return false, err
+	}
+	defer apiClient.Close() //nolint:errcheck
+	ping, err := apiClient.Ping(context.Background())
+	if err != nil {
+		return false, err
+	}
+	return ping.BuilderVersion == types.BuilderBuildKit, nil
+}
+
+// buildxConfigDir will look for correct configuration store path;
+// if `$BUILDX_CONFIG` is set - use it, otherwise use parent directory
+// of Docker config file (i.e. `${DOCKER_CONFIG}/buildx`)
+func buildxConfigDir(dockercfg *configfile.ConfigFile) string {
+	if buildxConfig := os.Getenv("BUILDX_CONFIG"); buildxConfig != "" {
+		return buildxConfig
+	}
+	return filepath.Join(filepath.Dir(dockercfg.Filename), "buildx")
+}
+
+// buildxDriver returns the build driver being used for the build command
+func buildxDriver(dockercfg *configfile.ConfigFile, buildArgs []string) string {
+	driver := "error"
+	configDir := buildxConfigDir(dockercfg)
+	if _, err := os.Stat(configDir); err != nil {
+		return driver
+	}
+	builder := buildxBuilder(buildArgs)
+	if len(builder) == 0 {
+		// if builder not defined in command, seek current in buildx store
+		// `${DOCKER_CONFIG}/buildx/current`
+		fileCurrent := path.Join(configDir, "current")
+		if _, err := os.Stat(fileCurrent); err != nil {
+			return driver
+		}
+		// content looks like
+		// {
+		//   "Key": "unix:///var/run/docker.sock",
+		//   "Name": "builder",
+		//   "Global": false
+		// }
+		rawCurrent, err := ioutil.ReadFile(fileCurrent)
+		if err != nil {
+			return driver
+		}
+		// unmarshal and returns `Name`
+		var obj map[string]interface{}
+		if err = json.Unmarshal(rawCurrent, &obj); err != nil {
+			return driver
+		}
+		if n, ok := obj["Name"]; ok {
+			builder = n.(string)
+			// `Name` will be empty if `default` builder is used
+			// {
+			//   "Key": "unix:///var/run/docker.sock",
+			//   "Name": "",
+			//   "Global": false
+			// }
+			if len(builder) == 0 {
+				builder = "default"
```

Comment:
Can’t we just
```suggestion
				return "docker"
```
here?

Reply:
Sure we can, just wanted to be explicit on this special case where builder name can be empty.